File: Kotlin/Aula-Kotlin-24-08-22/src/main/kotlin/classesabstratas/Cachorro.kt
package classesabstratas

class Cachorro(especie: String,
               nomeCientifico: String,
               habitat: String,
               carnivoro: Boolean,
               sexo: String,
               idade: Int,
               val nome: String
): Animal(especie, nomeCientifico, habitat, carnivoro, sexo, idade
) {
    override fun correr() {
        println("Correndo como um dog")
    }

    override fun dormir() {
        println("Dormir como um dog")
    }

    override fun andar() {
        println("Andar como um dog")
    }

    override fun comer() {
        println("Comer como dog")
    }

    override fun emitirSom() {
        println("Latir")
    }
}

File: JavaScript-TypeScript/Ex-JS-TS-06-09-22/src/cachorro.ts
import { Animal } from "./animal";

export class Cachorro extends Animal{




    seMovimentar() {
        console.log("Se movimentar como um cachorro")
    }


    comer() {
        console.log("Comer como um cachorro")
    }


    transmitirSom() {
        console.log("Au au")
    }

}